File: gencode/python/udmi/schema/monitoring.py
"""Generated class for monitoring.json"""
from .monitoring_metric import Monitoringmetric


class Monitoring:
  """Generated schema class"""

  def __init__(self):
    self.metric = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = Monitoring()
    result.metric = Monitoringmetric.from_dict(source.get('metric'))
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = Monitoring.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.metric:
      result['metric'] = self.metric.to_dict() # 4
    return result

File: gencode/java/udmi/schema/Monitoring.java

package udmi.schema;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Monitoring
 * <p>
 * Output from UDMIS monitoring
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "metric"
})
public class Monitoring {

    /**
     * Monitoring metric
     * <p>
     * One metric
     * 
     */
    @JsonProperty("metric")
    @JsonPropertyDescription("One metric")
    public MonitoringMetric metric;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.metric == null)? 0 :this.metric.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof Monitoring) == false) {
            return false;
        }
        Monitoring rhs = ((Monitoring) other);
        return ((this.metric == rhs.metric)||((this.metric!= null)&&this.metric.equals(rhs.metric)));
    }

}
